src/ml_board/frontend/Dashboard/src/redux/status/statusSlice.ts

```diff
@@ -55,7 +57,14 @@ export const statusSlice = createSlice({
     upsertMetricOrLoss: (state, { payload }: PayloadAction<string[]>) => {
       state.metric_loss.push(...payload);
     }
-  }
+  }, extraReducers(builder) {
+    builder.addCase(upsertCharts, (state, { payload }) => {
+      if (!state.color_map.hasOwnProperty(payload[0].exp_id)) {
+        // one can only hope that it doesn't produce a blue blue blue blue blue... pattern :')
+        state.color_map[payload[0].exp_id] = `#${Math.floor(Math.random() * 16777215).toString(16)}`;
```

Comment:
we had a color selection function previously defined that was deterministic. Why did we get rid of the original one?

Reply:
Well this is the old one, and I don't see how this is deterministic? 🤔
```js
function getRandomColor() {
    let letters = '0123456789ABCDEF'.split('');
    let color = '#';
    for (let i = 0; i < 6; i++) {
        color += letters[Math.floor(Math.random() * 16)];
    }
    return color;
}
```
but even with the one line that we have right now, I can after generating the color check if it's already there and create a new one? 
But that still doesn't grantee that the colors would look unique to our eyes (z.B. #eb4034 #cc3025 these are different numbers but to my eye the same color?! 😵‍💫)

Reply:
there was even another one, that I wrote that was older. The colors were selected such that they were maximally different to the human eye :laughing:

Reply:
Unfortunately I can't find it 🙁 but if so then it's exactly what we need!

maybe for now we create an issue and hopefully we find it later or even re-implement it later on.